lib/ui/compositing/scene.cc

```diff
@@ -126,15 +122,36 @@ void Scene::RasterizeToImage(uint32_t width,
 
   auto image = CanvasImage::Create();
   auto dl_image = CreateDeferredImage(
-      dart_state->IsImpellerEnabled(), layer_tree_, width, height,
-      std::move(snapshot_delegate), std::move(raster_task_runner),
-      std::move(unref_queue));
+      dart_state->IsImpellerEnabled(),
+      BuildLayerTree(width, height, pixel_ratio), std::move(snapshot_delegate),
+      std::move(raster_task_runner), std::move(unref_queue));
   image->set_image(dl_image);
   image->AssociateWithDartWrapper(raw_image_handle);
 }
 
-std::shared_ptr<flutter::LayerTree> Scene::takeLayerTree() {
-  return std::move(layer_tree_);
+std::unique_ptr<flutter::LayerTree> Scene::takeLayerTree(int64_t view_id) {
```

Comment:
Could this method accept `width`, `height`, and `pixel_ratio` values so that looking up the view's metrics is done by `RuntimeController::Render`?